Rework user guide introduction

alexisthual opened the pull request:
In line with what was discussed during the last core dev meeting and summarized in #3379, this PR partially rewrites the introduction section of the user guide.

In short, it:
- move install instructions to the Quickstart section (maybe we should remove the `venv` / `conda` part)
- removes most `sklearn` and Python related content (it only keeps links to these)
- moves the section of advanced ML applications to neuroimaging at the bottom
- entices users to discover the API reference section through a simple exercise
- gives a curated list of 8 examples to go through as a new-comer
- describes the user guide's targeted audience in the user guide's main page
- indices sections, subsections, etc using 1.2.3 notations

alexisthual commented:
Useful links:
- [install subsection in readme](https://output.circle-artifacts.com/output/job/8fa99f0c-05e5-467e-8594-6d8273224797/artifacts/0/dev/quickstart.html#install)
- [main user guide page](https://output.circle-artifacts.com/output/job/8fa99f0c-05e5-467e-8594-6d8273224797/artifacts/0/dev/user_guide.html)
- [user guide intro section](https://output.circle-artifacts.com/output/job/8fa99f0c-05e5-467e-8594-6d8273224797/artifacts/0/dev/introduction.html)

ymzayek commented on doc/introduction.rst:
```diff
@@ -10,501 +10,326 @@ Introduction
+====================
 
-.. topic:: **Why use nilearn?**
+``nilearn`` is a package that makes is easy to use advanced machine learning techniques so as to analyze data acquired with MRI machines.
```
```suggestion
``nilearn`` is a package that makes it easy to use advanced machine learning techniques to analyze data acquired with MRI machines.
```

ymzayek commented on doc/introduction.rst:
```diff
@@ -10,501 +10,326 @@ Introduction
-    :ref:`VBM <sphx_glr_auto_examples_02_decoding_plot_oasis_vbm.py>` data.
+For machine learning experts, the value of ``nilearn`` can be seen as
+domain-specific **feature engineering** construction, that is, shaping
+neuroimaging data into a feature matrix well suited to statistical learning.
```
```suggestion
neuroimaging data into a feature matrix well suited for statistical learning.
```

ymzayek replied:
```suggestion
neuroimaging data into a feature matrix well suited for statistical learning.
```

ymzayek commented on doc/introduction.rst:
```diff
@@ -10,501 +10,326 @@ Introduction
 
-Python for NeuroImaging, a quick start
-==========================================
+``nilearn`` comes in with some data that are commonly used in neuroimaging.
+For instance, it comes with volumic template images of brains such as MNI:
```
```suggestion
``nilearn`` comes with some data commonly used in neuroimaging.
For instance, it comes with volumetric template images of brains such as MNI:
```

alexisthual replied:
Haha, thanks, I always get volume/volumetric/surface wrong :smile:

ymzayek commented on doc/introduction.rst:
```diff
@@ -10,501 +10,326 @@ Introduction
+--------------------------------
 
-    .. note::
+In the last command, you just made use of 2 ``nilearn`` modules: the :mod:`nilearn.datasets`
+and the :mod:`nilearn.plotting` modules.
```
```suggestion
In the last command, you just made use of 2 ``nilearn`` modules: :mod:`nilearn.datasets`
and :mod:`nilearn.plotting`.
```

ymzayek commented on doc/introduction.rst:
```diff
@@ -10,501 +10,326 @@ Introduction
-    array([5, 4, 8, 8, 4, 9, 0, 8, 9, 8])
-    >>> labels[-10:]    # The actual labels
-    array([5, 4, 8, 8, 4, 9, 0, 8, 9, 8])
+We give a non-exhaustive list of important such applications.
```
```suggestion
We give a non-exhaustive list of such important applications.
```

ymzayek reviewed (commented):
@alexisthual great start! I just made some small suggestions.

alexisthual commented on README.rst:
```diff
@@ -34,14 +34,68 @@ Important links
+
+    python -m pip install -U nilearn
+
+Development version
+-------------------
+
+In order to access the development version of nilearn, simply clone and go to the repo:
+
+.. code-block:: bash
+
+    git clone https://github.com/nilearn/nilearn.git
+    cd nilearn
+
+Install the package in the proper conda environment with
+
+.. code-block:: bash
+
+    python -m pip install -e '.[dev]'
```
Yes that seems like a much better idea!
Do you think we should also move the part related to environments elsewhere?

ymzayek replied:
For now let's keep it there and then we can see if there's a more appropriate place to put it. But if there are other suggestions I say go for it!

alexisthual commented:
In the last commit, I tried to integrate all comments made during the last nilearn core dev meeting:
- added a link to the Handbook of fMRI data analysis
- removed the paragraph featuring Jupyter notebooks and IPython sessions, but kept it in a bullet point dedicated to python beginners at the bottom of the introduction page

I think this PR is ready for review :slightly_smiling_face:

ymzayek reviewed (approved):
Hi @alexisthual this LGTM! Can you add a whatsnew entry?

bthirion commented on doc/introduction.rst:
```diff
@@ -10,501 +10,327 @@ Introduction
-    :target: auto_examples/01_plotting/plot_demo_glass_brain.html
-    :align: center
-    :scale: 60
+    <div class="sphx-glr-thumbcontainer" tooltip="A simple example showing how to load an existing Nifti file and use basic nilearn functionaliti...">
```
```suggestion
    <div class="sphx-glr-thumbcontainer" tooltip="A simple example showing how to load an existing Nifti file and use basic nilearn functionalities">
```

bthirion commented on doc/introduction.rst:
```diff
@@ -10,501 +10,327 @@ Introduction
-              popular neuroimaging tools do not handle int64 Nifti images, so if
-              you build Nifti images directly from Numpy arrays it is recommended
-              to specify a smaller integer type, for example::
+    <div class="sphx-glr-thumbcontainer" tooltip="This example shows manual steps to create and further modify an ROI spatial mask. They represen...">
```
unended sentence.

alexisthual replied:
This is because that's what the tooltip looks like for these examples in the examples gallery :)

bthirion commented on doc/introduction.rst:
```diff
@@ -10,501 +10,327 @@ Introduction
+.. raw:: html
 
-:Image processing:
+    <div class="sphx-glr-thumbcontainer" tooltip="Here, we will go through a full step-by-step example of fitting a GLM to experimental data and ...">
```
Same here.

bthirion commented on doc/introduction.rst:
```diff
@@ -10,501 +10,327 @@ Introduction
+.. raw:: html
 
-  .. hlist::
+    <div class="sphx-glr-thumbcontainer" tooltip="This example compares different kinds of functional connectivity between regions of interest : ...">
```
Same here

bthirion commented on doc/introduction.rst:
```diff
@@ -10,501 +10,327 @@ Introduction
 
-Finding help
--------------
+**Measuring generalization scores**
```
This part sounds a bit weird.

alexisthual replied:
Agreed ; I got rid of the title altogether.

bthirion reviewed (commented):
Thx. Please find a few comments here.

ymzayek commented on doc/introduction.rst:
```diff
@@ -10,501 +10,329 @@ Introduction
+Besides, you can find help on :neurostars:`neurostars <>` for questions
+related to ``nilearn`` and to computational neuroscience in general.
+Finally, the ``nilearn`` team organizes weekly :ref:`office hours <quickstart>`.
+We can also be reached on :ref:`github <https://github.com/nilearn/nilearn/issues>`
```
```suggestion
We can also be reached on :nilearn-gh:`github <issues>`
```

^that should work. When in doubt you can always do it this way:
 ```
`github <https://github.com/nilearn/nilearn/issues>`_
```

bthirion commented:
I don't understand the builddoc failure... docs seem to have built correctly.
I don't find the artifacts...

ymzayek commented:
@alexisthual thanks! The doc build passed now and I think this is ready to merge. 

@bthirion the failure was just a syntax error in a reference. The artifacts can only be viewed after a successful build, which is the case now. If you scroll down the list of checks you can see a CircleCI workflow called host_and_deploy_doc where you can view the artifacts...linking here https://github.com/nilearn/nilearn/pull/3380/checks?check_run_id=9734785256

bthirion reviewed (approved):
This looks great, thx !

ymzayek commented:
Merging
